archive_viewer/mixins/traces_table.py

```diff
@@ -162,73 +169,96 @@ def import_csv(self) -> None:
     def __init__(self, parent: QObject) -> None:
         super().__init__(parent)
         self.setWindowTitle("Formula Input")
 
         # Create the layout for the dialog
         layout = QVBoxLayout(self)
-
         # Create the QLineEdit for formula input
         self.field = QLineEdit(self)
+        self.curveModel = self.parent().parent().curves_model
+        self.pv_list = QTableView(self)
+        #We're going to copy the list of PVs from the curve model. We're also not going to allow the user to make edits to the list of PVs
+        self.pv_list.setModel(self.curveModel)
+        self.pv_list.setEditTriggers(QAbstractItemView.EditTriggers(0))
+        self.pv_list.setMaximumWidth(1000)
+        header = self.pv_list.horizontalHeader()
+        header.setSectionResizeMode(0, QHeaderView.Stretch)
+        for i in range(1, self.curveModel.columnCount() - 1):
+            # Hide all columns that arent useful, but keep one left over to add a button to
+            self.pv_list.setColumnHidden(i, True)
+        insertButton = InsertPVDelegate(self.pv_list)
+        insertButton.button_clicked.connect(self.field.insert)
+        self.pv_list.setItemDelegateForColumn(self.curveModel.columnCount() - 1, insertButton)
+        layout.addWidget(self.pv_list)
         layout.addWidget(self.field)
 
-        # Define the list of calculator buttons
-        buttons = ["7", "8", "9", "+",
-                   "4", "5", "6", "-",
-                   "1", "2", "3", "*",
-                   "0", "(", ")", "/",
-                   ".", "PV", "Clear", "="]
+        self.index = self.parent().selected_index
+
+        # Define the list of calculator buttons.
+        # It's a bunch of preset buttons, but users can type other functions under math.
+        buttons = ["7",       "8",     "9",      "+",     "(",      ")",
+                   "4",       "5",     "6",      "-",    "^2", "sqrt()",
+                   "1",       "2",     "3",      "*",   "^-1",  "ln()",
+                   "0",       "e",    "pi",      "/", "sin()", "asin()",
+                   ".",   "abs()", "min()",      "^", "cos()", "acos()",
+                   "PV",  "Clear", "max()", "mean()", "tan()", "atan()"]
 
         # Create the calculator buttons and connect them to the input field
         grid_layout = QGridLayout()
         for i, button_text in enumerate(buttons):
             button = QPushButton(button_text, self)
-            row = i // 4
-            col = i % 4
+            row = i // 6
+            col = i % 6
             grid_layout.addWidget(button, row, col)
-
             # Connect the button clicked signal to the appropriate action
+            # PV currently does nothing, this is a remnant
+            # From when we would have the pv_list open in a new window
             if button_text == "PV":
-                button.clicked.connect(lambda _: self.field.insert("PV"))
+                # TODO: Either give a function for this button or replace it
```

Comment:
if you don't think you need the PV button I would remove it

Reply:
I did not consider this but I definitely could add it. No, it never added/hit the list of PVs. you may be thinking of the PV search tool which is one of the options in the custom dialogue box, one level above the formula dialogue box

Reply:
Weird. I don't know what I'm thinking of then. I really thought that was an idea you had messed around with and implemented. Oh well.

Not sure it's the most useful functionality for this button, could just cause confusion. Not sure what this button could do.
Is there a more useful function that we could include instead? Even just `^e`? Might need to move the button to a new place too